### ISSUE TITLE
[BUG] OptiGUI doesn't work, if container is opened via Spectator Mode

### ISSUE THREAD

**Issue by Vi-Tul** [opekope2/OptiGUI#76]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Expected Behavior

Spectator Mode is working the same way as Survival and Creative

### Actual Behavior

When container is opened in Spectator Mode, neither OptiGUI nor Optifine format textures apply

- If opened in Creative Mode (texture is using Optifine format):

![image](https://github.com/opekope2/OptiGUI/assets/92220631/6235f848-5825-4e8a-8977-7a26cdb3686a)

- If opened in Spectator Mode (now it's using generic_54 texture):

![image](https://github.com/opekope2/OptiGUI/assets/92220631/920ecda4-ffde-4897-a23a-129cf2f888e3)

### Steps To Reproduce

Open any container that is using Opifine or OptiGUI format using Spectator Mode

### Loader and its version

Quilt loader 0.20.2 1.20.1

### OptiGUI version

2.2.0-alpha.1

### Behavior difference

- [ ] Behavior is different from OptiGUI documentation
- [ ] Behavior is different from OptiFine
- [X] Behavior is different from expected (or straight up doesn't work)

**Comment by opekope2** [comment]
Indeed, that doesn't work. It's because of [how Fabric API works](https://github.com/FabricMC/fabric/blob/3f301502811f0ab410f045d065b6f98d5b630537/fabric-events-interaction-v0/src/client/java/net/fabricmc/fabric/mixin/event/interaction/client/ClientPlayerInteractionManagerMixin.java#L98). I may need to hook it myself. Ironic.

**Comment by mMAK141** [comment]
hey ViTulyankin
